src/ape/api/contracts.py

```diff
@@ -0,0 +1,303 @@
+        def get_name(abi: ABI):
+            return abi.name
+
+        try:
+            if attr_name in map(get_name, self._contract_type.events):
+                selected_abis = [
+                    abi for abi in self._contract_type.events if get_name(abi) == attr_name
+                ]
+                return ContractEvent(  # type: ignore
+                    provider=self.provider,
+                    address=self.address,
+                    abis=selected_abis,  # type: ignore
+                )
+
+            elif attr_name in map(get_name, self._contract_type.calls):
+                selected_abis = [
+                    abi for abi in self._contract_type.calls if get_name(abi) == attr_name
+                ]
+                return ContractCallHandler(  # type: ignore
+                    provider=self.provider,
+                    address=self.address,
+                    abis=selected_abis,
+                )
+
+            elif attr_name in map(get_name, self._contract_type.transactions):
+                selected_abis = [
+                    abi for abi in self._contract_type.transactions if get_name(abi) == attr_name
+                ]
+                return ContractTransactionHandler(  # type: ignore
+                    provider=self.provider,
+                    address=self.address,
+                    abis=selected_abis,
+                )
+
+        except Exception as e:
+            raise AttributeError from e
+
+        raise AttributeError(f"{self.__class__.__name__} has no attribute '{attr_name}'")
+
+
+@dataclass
+class ContractContainer:
+    _provider: ProviderAPI
+    _contract_type: ContractType
+
+    def __repr__(self) -> str:
+        return f"<{self._contract_type.contractName}>"
+
+    def at(self, address: str) -> ContractInstance:
+        return ContractInstance(  # type: ignore
+            _address=address,
+            _provider=self._provider,
+            _contract_type=self._contract_type,
+        )
+
+    @property
+    def _deployment_bytecode(self) -> bytes:
+        if (
+            self._contract_type.deploymentBytecode
+            and self._contract_type.deploymentBytecode.bytecode
+        ):
+            return to_bytes(hexstr=self._contract_type.deploymentBytecode.bytecode)
+
+        else:
+            return b""
+
+    @property
+    def _runtime_bytecode(self) -> bytes:
+        if self._contract_type.runtimeBytecode and self._contract_type.runtimeBytecode.bytecode:
+            return to_bytes(hexstr=self._contract_type.runtimeBytecode.bytecode)
+
+        else:
+            return b""
+
+    def __call__(self, *args, **kwargs) -> TransactionAPI:
+        constructor = ContractConstructor(  # type: ignore
+            abi=self._contract_type.constructor,
+            provider=self._provider,
+            deployment_bytecode=self._deployment_bytecode,
+        )
+        return constructor.encode(*args, **kwargs)
+
+
+def Contract(
```

Comment:
I think it's a little confusing to have three different ways to import `Contract`:

```
In [1]: from ape import Contract as C1
In [2]: from ape.api import Contract as C2
In [3]: from ape.api.contracts import Contract as C3
In [4]: C1 == C2, C2 == C3
Out[4]: (False, True)
```

If you want "C1" above to be the real class used by end users, maybe C2 and C3 should be named `_Contract` in order to hide that implementation detail.

Reply:
Fair. This is what we want.